### ISSUE TITLE
Hybrid app - Getting duplicate notifications for the Expensify statement 

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#51918]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Needs Reproduction
**Reproducible in production?:** Needs Reproduction
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:** https://expensify.slack.com/archives/C049HHMV9SM/p1730491509556909
**Issue reported by:** @AndrewGable 
**Slack conversation** (hyperlinked to channel name): [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1730491509556909)

## Action Performed:

1. Login to the HybridApp.
2. Set up and enable Expensify statement notifications.

## Expected Result:
One notification from Concierge regarding statement 
## Actual Result:
Duplicate notifications received for same month
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [X] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  

<img src="https://github.com/user-attachments/assets/9dad4a82-a3c9-4705-8007-2c2a18c73426" width="40%" height="40%">

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by jliexpensify** [comment]
Sorry, the weekend then was OOO so I missed this. I don't have an iOS device so going to post in the #quality (?) channel. I'm thinking it might be `Internal`.

**Comment by jliexpensify** [comment]
[Posted in #quality](https://expensify.slack.com/archives/C05LX9D6E07/p1730856170541999)

**Comment by jliexpensify** [comment]
Not overdue, just added labels

**Comment by jliexpensify** [comment]
Waiting on an internal engineer

**Comment by jliexpensify** [comment]
Waiting for an internal engineer.

**Comment by muttmuure** [comment]
Does this happen when you're logged into NewApp and HybridApp at the same time? Or just HybridApp?

**Comment by AndrewGable** [comment]
HybridApp only

**Comment by jliexpensify** [comment]
Internal, waiting on an engineer

**Comment by AndrewGable** [comment]
Still happening BTW


![unnamed](https://github.com/user-attachments/assets/345eb3ca-9268-4846-80ad-836bc0cfe7d1)

**Comment by arosiclair** [comment]
I'll look into this when I get some cycles

**Comment by arosiclair** [comment]
I spent some time on this yesterday. I should be able to reproduce this in dev by just running the `ExpensifyWalletRegulatoryEmails`:

```
CreateJob
name: www-prod/ExpensifyWalletRegulatoryEmails
data: {"debugEmail": "test@email.com"}
```

I wasn't able to reproduce with NewDot so I may need to test with the HybridApp specifically.

**Comment by arosiclair** [comment]
Looks like this is affecting most old push notifications:

<img src="https://github.com/user-attachments/assets/dc6ed2c8-3039-45e6-95cd-2d48782a376b" width="300" />

**Comment by AndrewGable** [comment]
Agree - I think I saw it on most of them

**Comment by arosiclair** [comment]
No progress on this. HybridApp and push notification dev aren't in a good spot right now so this probably has to wait until that gets better.

**Comment by arosiclair** [comment]
Alright I was able to reproduce this issue with unit tests. The root problem is in the logic for [sendReportCommentNotifications](https://github.com/Expensify/Web-Expensify/blob/dbee24c29083a6c0ba02df116316875fd38fde82/lib/MobilePushNotifications.php#L64). For concierge chats, both `$shouldSendToNewDot` and `$shouldSendToOldDot` are true so we end up sending push notifications twice.

With the HybridApp in play, I don't think choosing the destination/payload based on the report type makes sense anymore so instead this should probably be completely controlled by the `tryNewDot` NVP. So I think I'll update [sendPushNotification](https://github.com/Expensify/Web-Expensify/blob/dbee24c29083a6c0ba02df116316875fd38fde82/lib/MobilePushNotifications.php#L412) to take both the old and new payloads as params and then it can choose which to use accordingly.

**Comment by arosiclair** [comment]
Posted the PR

**Comment by arosiclair** [comment]
Duplicate Concierge push notifications are now fixed but [this](https://github.com/Expensify/App/issues/51918#issuecomment-2541940827) is still happening since those aren't Concierge messages. I'll look into it soon.

**Comment by arosiclair** [comment]
I wasn't able to reproduce the duplicate transaction notifications using unit tests so it looks unrelated to `MobilePushNotifications` logic. Instead, I think there's probably an issue with [TransactionNotifications::sendNotification](https://github.com/Expensify/Web-Expensify/blob/31b30fd0cdcb9aeafc806efddbe5a76f4e2ee0b5/lib/Marqeta/TransactionNotifications.php#L141). We might be accidentally calling that twice in response to Marqeta webhook notifications. cc @madmax330 in case you have any ideas.

**Comment by arosiclair** [comment]
Since this is a separate issue, I reported it in `#quality` [here](https://expensify.slack.com/archives/C05LX9D6E07/p1737752445806269) so we can open a new GH and get it re-prioritized. Closing this out
